pkgs/applications/science/math/mxnet/default.nix

```
@@ -1,50 +1,63 @@
-{ config, stdenv, lib, fetchurl, bash, cmake
-, opencv3, gtest, blas, perl
-, cudaSupport ? config.cudaSupport or false, cudatoolkit, nvidia_x11
-, cudnnSupport ? cudaSupport, cudnn
+{ config
+, stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, gtest
+, opencv3
+, blas
+, perl
+, cudaSupport ? config.cudaSupport or false
+, cudatoolkit
+, nvidia_x11
+, cudnnSupport ? cudaSupport
+, cudnn
 }:
 
 assert cudnnSupport -> cudaSupport;
 
 stdenv.mkDerivation rec {
   pname = "mxnet";
-  version = "1.6.0";
+  version = "1.8.0";
 
-  src = fetchurl {
-    url = "https://github.com/apache/incubator-mxnet/releases/download/${version}/apache-mxnet-src-${version}-incubating.tar.gz";
-    sha256 = "1vvdb7pfh63kb9fzs6gqp95q550a3ck4cj9mqxlk9wwhkh30dsq1";
+  src = fetchFromGitHub {
+    owner = "apache";
+    repo = "incubator-mxnet";
+    rev = version;
+    sha256 = "sha256-/IwFMEOStqkptLrCpi6UiD0IOh/Cgz+7o4alqUXWIis=";
+    fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [ cmake perl ];
-
-  buildInputs = [ opencv3 gtest blas.provider ]
-              ++ lib.optionals cudaSupport [ cudatoolkit nvidia_x11 ]
-              ++ lib.optional cudnnSupport cudnn;
-
-  cmakeFlags =
-    [ "-DUSE_MKL_IF_AVAILABLE=OFF" ]
-    ++ (if cudaSupport then [
-      "-DUSE_OLDCMAKECUDA=ON"  # see https://github.com/apache/incubator-mxnet/issues/10743
-      "-DCUDA_ARCH_NAME=All"
-      "-DCUDA_HOST_COMPILER=${cudatoolkit.cc}/bin/cc"
-    ] else [ "-DUSE_CUDA=OFF" ])
-    ++ lib.optional (!cudnnSupport) "-DUSE_CUDNN=OFF";
-
+  # Build against the system version of OpenMP.
   postPatch = ''
-    substituteInPlace 3rdparty/mkldnn/tests/CMakeLists.txt \
-      --replace "/bin/bash" "${bash}/bin/bash"
-
-    # Build against the system version of OpenMP.
-    # https://github.com/apache/incubator-mxnet/pull/12160
     rm -rf 3rdparty/openmp
   '';
 
+  nativeBuildInputs = [ cmake gtest perl ];
+
+  buildInputs = [ opencv3 blas.provider ]
+    ++ lib.optionals cudaSupport [ cudatoolkit nvidia_x11 ]
+    ++ lib.optional cudnnSupport cudnn;
+
+  cmakeFlags = [
+    "-DUSE_MKL_IF_AVAILABLE=OFF"
+  ]
+  ++ (if cudaSupport then [
+    "-DUSE_OLDCMAKECUDA=ON" # see https://github.com/apache/incubator-mxnet/issues/10743
+    "-DCUDA_ARCH_NAME=All"
+    "-DCUDA_HOST_COMPILER=${cudatoolkit.cc}/bin/cc"
+  ] else [ "-DUSE_CUDA=OFF" ])
+  ++ lib.optional (!cudnnSupport) "-DUSE_CUDNN=OFF";
+
   postInstall = ''
     rm "$out"/lib/*.a
   '';
 
   meta = with lib; {
-    description = "Lightweight, Portable, Flexible Distributed/Mobile Deep Learning with Dynamic, Mutation-aware Dataflow Dep Scheduler";
+    description = ''
```

description shouldn't be multiline. Please use longDescription instead.

```suggestion
    longDescription = ''
```